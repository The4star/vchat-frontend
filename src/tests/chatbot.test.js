import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Chatbot from '../components/chatbot/chatbot.component';
// import Message from '../components/message/message.component'
 const wrapper = shallow(<Chatbot />)

it('renders the chatbot component', () => {
  expect(wrapper).toMatchSnapshot()
})

describe('function testing', () => {
  it('toggles the bot from hidden to showing', () => {
    expect(wrapper.instance().state.hidden).toEqual(true)
    wrapper.instance().toggleBot()
    expect(wrapper.instance().state.hidden).toEqual(false)
  })

  it('performs a text query', () => {
    wrapper.instance().textQuery('hi')
    expect(wrapper.instance().state.messages.length).toEqual(1)
    // console.log(wrapper.instance().state.messages)
  })
  
  it('renders messages', () => {
   expect(wrapper.instance().renderMessages([ { speaker: 'me', msg: 'hi' }  ])).toHaveLength(1)
   expect(wrapper.instance().renderMessages([ { speaker: 'me', msg: 'hi' }, { speaker: 'vchat', msg: 'hi, im vchat' }  ])).toHaveLength(2)
  })
})