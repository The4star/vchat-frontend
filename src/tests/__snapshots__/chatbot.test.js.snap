// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the chatbot component 1`] = `ShallowWrapper {}`;
